### DISCUSSION TITLE
A help needed question on collection mapping

### DISCUSSION THREAD

**Discussion by mdsardar** [mapstruct/mapstruct#2565] [Q&A]
Hi All, 

I've recently started to use mapstruct, while coding I stuck with a scenario. In order to solve the ambiguity between the below default methods, I'm trying to use the "qualifiedBy" on a List<

   ```
@Mapping(qualifiedBy = LineTestBO). Line => 1
    List<ReturnABO> toCaptureLineItemsBOs(List<LineDTO> lineDTO);

    @Named("LineTestBO")
    default ReturnABO map(LineDTO lineDTO) {
        if (lineDTO.getCurrency() != null && lineDTO.getNationalPermit() != null &&
            lineDTO.getAmount() != null && lineDTO != null)
            return this.toBO(lineDTO);
        return null;
    }

     default returnABO toBO(LineDTO lineDTO)  {
     // To do here
}
```

But Line 1 shows the error as it needs to "target" to be specified. I'm not sure what should be the target here since Line<ReturnABO> is a collection object.  Even If I don't use the @Mapping the mapstuct generates the mapper implementation. I read the mapstuct documentation but could not follow much for this scenario.  How the Named annotation can be used on List<ReturnBO> to explicity denote that this is name to be used ? Can someone help me? Thanks.

**Comment by filiphr** [comment]
Hey @mdsardar. I see that you have also raised the same question on StackOverflow. I've provided an answer to it [here](https://stackoverflow.com/a/68989791/1115491). Please do not cross post, or at least link to a StackOverflow question when you ask it here as well, it will help people not to do double work.

**Reply by mdsardar** [reply]
Hey @filiphr, Thanks for your response. I raised here first then in stack since couldn't hear anything from here, thought to reach a broad forum .ie. Stack. If this is the procedure I'm ok to follow.
